internal/server/server.go

```diff
@@ -199,9 +214,71 @@ func recordsToStartupEventInfo(records []*neo4j.Record, mcpVersion string) analy
 }
 
 // Stop gracefully stops the server
-func (s *Neo4jMCPServer) Stop() error {
+func (s *Neo4jMCPServer) Stop(ctx context.Context) error {
 	slog.Info("Stopping Neo4j MCP Server...")
-	// Currently no cleanup needed - the MCP server handles its own lifecycle
-	// Database service cleanup is handled by the caller (main.go)
+	if s.httpServer != nil {
+		if err := s.httpServer.Shutdown(ctx); err != nil {
+			slog.Error("Error shutting down HTTP server", "error", err)
+			return err
+		}
+	}
+	slog.Info("Neo4j MCP Server stopped.")
 	return nil
 }
+
+func (s *Neo4jMCPServer) StartHTTPServer() error {
+	addr := fmt.Sprintf("%s:%s", s.config.HTTPHost, s.config.HTTPPort)
+	serverURL := fmt.Sprintf("http://%s", addr)
```

Comment:
do we want to have the path `/mcp` included or configurable?

Reply:
I will double check but I think it’s already included in the SDK path once we use the MCP server handler as the base. Which allows us to add more endpoints like /health at the root level for CI.